src/requests/typings.ts

```diff
@@ -0,0 +1,64 @@
+import * as runtypes from 'runtypes'
+
+const MethodValidator = runtypes.Union(
+  runtypes.Literal('GET'),
+  runtypes.Literal('PATCH'),
+  runtypes.Literal('HEAD'),
+  runtypes.Literal('POST'),
+  runtypes.Literal('DELETE'),
+  runtypes.Literal('OPTIONS'),
+  runtypes.Literal('PUT')
+)
+
+const PathValidator = runtypes.String.withConstraint((s) => s.startsWith('/'), {
```

Comment:
btw, aren't URLs actually including protocol & domain?

Reply:
URLs yes, but this is a path. IIRC we decide to only use the canonical path (without protocol and domain) so that it is independent from load balancers, reverse proxies, etc.?

Reply:
I wasn't clear enough in the comment. That part you mentioned  is fine - but I'm thinking in terms of public interface of this. Basically wondering where we should be concerned with splitting path from URL. Maybe it makes sense to handle this in the `POST` handler for signing requests. We take the full URL there and send the path for signing. Otherwise, will we know in SDK where to send the request if we also only require a path there.